src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -1092,7 +1091,12 @@ bool isLambdaConversion(BoundExpression expr)
                 }
             }
 
-            return CreateParamArrayArgument(syntax, paramArrayType, arrayArgs, _compilation, this);
+            if (paramArrayType.IsSZArray())
+            {
+                return CreateParamArrayArgument(syntax, paramArrayType, arrayArgs, _compilation, this);
```

Comment:
>CreateParamArrayArgument [](http://example.com/codeflow?start=23&length=24)

There are several more call sites for this method. Should they be adjusted as well? Also, would it make sense to move the logic that deals with param array to initial binding as we have done for default values?